tests/test_opal_api.py

```diff
@@ -0,0 +1,83 @@
+import json
+
+import pandas as pd
+import pytest
+from fastapi.testclient import TestClient
+
+from datahub import data as dt
+from datahub.main import app
+from datahub.opal import create_opal_frame
+
+client = TestClient(app)
+client.headers["Content-Type"] = "application/json"
```

Comment:
I'd turn this into a fixture and move it into `conftest.py`. Something like:
```suggestion
from datahub.opal import create_opal_frame

@pytest.fixture()
def client():
    from datahub.main import app

    client = TestClient(app)
    client.headers["Content-Type"] = "application/json"

    return client
```